lib/layer.js

```diff
@@ -65,9 +78,23 @@ Layer.prototype.handle_error = function handle_error(error, req, res, next) {
     return next(error)
   }
 
+  req.layerStack = req.layerStack || []
```

Comment:
I would recommend against creating a new array for every request if diagnostics_channel is not enabled.

Reply:
Fair. Seemed like a possibly useful feature outside of the diagnostics_channel stuff itself, but sounds like there's already another layer-tracking thing in discussion, so I'll take a closer look at that. :)

Reply:
May also need to consider what the expected behavior is when different versions of this module are mixed together.